File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    var answer = "";
    const temp = [...goal];
    
    while(temp.length){
        if(cards1[0]===temp[0]){
            answer += cards1.shift();
        }else if(cards2[0] === temp[0]){
            answer += cards2.shift();
        } else{
            break;
        }
        temp.shift();
    }
    
    return answer === goal.join("") ? "Yes" : "No";
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    
    new_goal = deque(goal)
    card_queue1 = deque(cards1)
    card_queue2 = deque(cards2)
    
    
    while new_goal:
        
        if( card_queue1 and card_queue1[0] == new_goal[0]):
            card_queue1.popleft()
            new_goal.popleft()
        elif card_queue2 and card_queue2[0] == new_goal[0]:
            card_queue2.popleft()
            new_goal.popleft()
        else :
            break;
            
    if new_goal :
        return "No"
    else:
        return "Yes"
